src/Stratis.Bitcoin.Tests/BlockPulling2/BlockPullerBehaviorTests.cs

```diff
@@ -0,0 +1,118 @@
+﻿using System;
+using Moq;
+using Stratis.Bitcoin.BlockPulling2;
+using Stratis.Bitcoin.Configuration.Logging;
+using Stratis.Bitcoin.Interfaces;
+using Xunit;
+
+namespace Stratis.Bitcoin.Tests.BlockPulling2
+{
+    public class BlockPullerBehaviorTests
+    {
+        private readonly BlockPullerBehavior behavior;
+
+        public BlockPullerBehaviorTests()
+        {
+            var puller = new Mock<IBlockPuller>();
+
+            var ibdState = new Mock<IInitialBlockDownloadState>();
+            ibdState.Setup(x => x.IsInitialBlockDownload()).Returns(() => true);
+
+            var loggerFactory = new ExtendedLoggerFactory();
+            loggerFactory.AddConsoleWithFilters();
+
+            this.behavior = new BlockPullerBehavior(puller.Object, ibdState.Object, loggerFactory);
+        }
+
+        [Fact]
+        public void InitializesWithDefaultValues()
+        {
+            Assert.Equal(BlockPullerBehavior.SamplelessQualityScore, this.behavior.QualityScore);
+            Assert.Equal(0, this.behavior.SpeedBytesPerSecond);
+            Assert.Null(this.behavior.Tip);
+        }
+
+        [Fact]
+        public void QualityScoreCanGoToMinimum()
+        {
+            this.behavior.Penalize(10, 10);
```

Comment:
It is not very obvious what are you doing here. It would be nice if the test did have a summary.

Reply:
Fixed